### ISSUE TITLE
Price is right game

### ISSUE THREAD

**Issue by oceans404** [NillionNetwork/nada-by-example#10]
### Description
Create a [“Price is Right” style game](https://en.wikipedia.org/wiki/The_Price_Is_Right) where multiple parties guess the price of an item, and the winner is the party whose guess is closest to the actual price without exceeding it.

### Use Case
This example can be extended to other types of competitive scenarios, such as auctions or bidding systems, making it valuable for developers interested in game theory, competitive pricing models, or privacy-preserving applications.

### Specific Requirements
Inputs:
- The actual price of the item as a SecretInteger
- The SecretInteger guesses from multiple parties.

Outputs:
- The guess that is closest to the actual price without exceeding it.
- The identification of the winning party.

Edge Cases:
- If all guesses exceed the actual price, the output should indicate that there is no winner.
- Handle ties where multiple guesses are the closest without exceeding the actual price.
- Consider cases where multiple guesses are equidistant from the actual price, both above and below.

**Comment by ysongh** [comment]
Hello,

I am interested in working on this part. I am confused about this output: 'The guess that is closest to the actual price without exceeding it.' Does this mean the player's input cannot be greater than the price?

**Comment by oceans404** [comment]
> Hello,
> 
> I am interested in working on this part. I am confused about this output: 'The guess that is closest to the actual price without exceeding it.' Does this mean the player's input cannot be greater than the price?

Yes, if a player's guess is higher than the actual price, they can't be the winner. Check out the game logic in this article: https://en.wikipedia.org/wiki/The_Price_Is_Right

**Comment by ysongh** [comment]
Thank You. I finish and made a pull request. https://github.com/NillionNetwork/nada-by-example/pull/38
